roles/ceph-osd/tasks/check_devices.yml

```diff
@@ -4,8 +4,9 @@
 # it should exist we rc=0 and don't do anything unless we do something like --force
 # As as a final word, I prefer to keep the partition check instead of running ceph-disk prepare with "failed_when: false"
 # I believe it's safer
+
 - name: check if the device is a partition or a disk
-  shell: "echo '{{ item }}' | egrep '/dev/(([a-z]{3,4}[0-9]$)|(cciss/c[0-9]{1}d[0-9]{1}p[0-9]$))'"
+  shell: "echo '{{ item }}' | egrep '/dev/[sd[a-z]{1,2}|hd[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p][0-9]{1,2}'"
```

Comment:
https://github.com/ceph/ceph-ansible/pull/412 updated with $ on end as you suggested.  So there are different families of device names, but the partition number in the regex is shared across all of them.

'/dev/[sd[a-z]{1,2}|hd[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p][0-9]{1,2}$'"

ok?

----- Original Message -----
> From: "Leseb" <notifications@github.com>
> To: "ceph/ceph-ansible" <ceph-ansible@noreply.github.com>
> Cc: "Ben England" <bengland@redhat.com>
> Sent: Wednesday, October 7, 2015 12:26:47 PM
> Subject: Re: [ceph-ansible] fix regex for partition names, enhance for NVM (#412)
> 
> >  - name: check if the device is a partition or a disk
> > -  shell: "echo '{{ item }}' | egrep
> > '/dev/(([a-z]{3,4}[0-9]$)|(cciss/c[0-9]{1}d[0-9]{1}p[0-9]$))'"
> > +  shell: "echo '{{ item }}' | egrep
> > '/dev/[sd[a-z]{1,2}|hd[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p][0-9]{1,2}'"
> 
> Hum shouldn't it be something more like:
> 
> `egrep '/dev/sd[a-z]{1,2}[0-9]$|hd ... ...`
> 
> Otherwise it looks like the condition is always true.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/ceph/ceph-ansible/pull/412/files#r41412326